dash/dash-renderer/src/observers/setProps.ts

```diff
@@ -0,0 +1,30 @@
+
+import {IStoreState} from '../store';
+import {updateProps, notifyObservers} from '../actions/index'
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    observer: ({dispatch, getState}) => {
+        const setProps = (updates: {}) => {
+            const {paths} = getState()
+            Object.entries(updates).forEach(([componentId, props]) => {
+                const componentPath = paths.strs[componentId];
+                dispatch(
+                    updateProps({
+                        props,
+                        itempath: componentPath
+                    })
+                );
+                dispatch(
+                    notifyObservers({id: componentId, props})
+                );
+            });
+        }
```

Comment:
The "observer" pattern in this redux context is very weird, it's meant to act as reducer with side effects.
I think this will be called on every callbacks and it should be only defined once. Maybe just move the definition on the window to store.ts in createApp instead would be better.

Reply:
Within this context:
```
private createAppStore = (reducer: any, middleware: any) => {
        this.__store = createStore(reducer, middleware);
        this.storeObserver.setStore(this.__store);
        this.setObservers();
    };
    ```